conftest.py

```diff
@@ -14,3 +20,48 @@ async def aiohatchet() -> AsyncGenerator[Hatchet, None]:
 @pytest.fixture(scope="session")
 def hatchet() -> Hatchet:
     return Hatchet(debug=True)
+
+
+@pytest.fixture()
+def worker(request: pytest.FixtureRequest):
+    example = cast(str, request.param)
+
+    command = ["poetry", "run", example]
+
+    logging.info(f"Starting background worker: {' '.join(command)}")
+    proc = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    # Check if the process is still running
+    if proc.poll() is not None:
+        raise Exception(f"Worker failed to start with return code {proc.returncode}")
+
+    def wait_for_log_message(pipe: BytesIO) -> None:
+        for line in iter(pipe.readline, b""):
+            log_line = line.decode().strip()
+            logging.info(log_line)  # Log the output
+            if "sending heartbeat" in log_line:
+                logging.info(f"Found target log message: {log_line}")
+                break
+
+    Thread(target=wait_for_log_message, args=(proc.stdout, ), daemon=True).start()
```

Comment:
Got rid of the `time.sleep` here in favor of waiting for `sending heartbeat` to show up - should help limit flakiness